Lib/fontTools/otlLib/builder.py

```diff
@@ -348,6 +348,76 @@ def build(self):
         # buildLookup_
         return self.buildLookup_(subtables)
 
+    def buildFormat2Subtable(self, ruleset, classdefs, chaining=True):
+        st = self.newSubtable_(chaining=chaining)
+        st.Format = 2
+        st.populateDefaults()
+
+        if chaining:
+            (
+                st.BacktrackClassDef,
+                st.InputClassDef,
+                st.LookAheadClassDef,
+            ) = [c.build() for c in classdefs]
+        else:
+            st.ClassDef = classdefs[1].build()
+
+        inClasses = classdefs[1].classes()
+
+        classSets = []
+        for _ in inClasses:
+            classSet = self.newRuleSet_(format=2, chaining=chaining)
+            classSet.populateDefaults()
```

Comment:
you've already called populateDefaults inside the `newRuleSet_` method so this second call is unneeded